Sources/PayPal/UIComponents/PayPalClient/PayPalClient.swift

```diff
@@ -0,0 +1,61 @@
+import UIKit
+
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+#if canImport(PayPalCheckout)
+@_implementationOnly import PayPalCheckout
+#endif
+
+public typealias PayPalCheckoutCompletion = (PayPalCheckoutResult) -> Void
+
+/// PayPal Paysheet to handle PayPal transaction
+public class PayPalClient {
+
+    private let config: CoreConfig
+    private let returnURL: String
+
+    lazy var paypalCheckoutConfig: CheckoutConfig = {
+        CheckoutConfig(
+            clientID: config.clientID,
+            returnUrl: returnURL,
+            environment: config.environment.toPayPalCheckoutEnvironment()
+        )
+    }()
+
+    public init(config: CoreConfig, returnURL: String) {
+        self.config = config
+        self.returnURL = returnURL
+    }
+
+    /// Present PayPal Paysheet and start a PayPal transaction
+    /// - Parameters:
+    ///   - orderID: the order ID for the transaction
+    ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
+    ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
+    public func start(orderID: String, presentingViewController: UIViewController? = nil, completion: PayPalCheckoutCompletion? = nil) {
+        setupPayPalCheckoutConfig(orderID: orderID, completion: completion)
+        Checkout.start(presentingViewController: presentingViewController)
+    }
+
+    func setupPayPalCheckoutConfig(orderID: String, completion: PayPalCheckoutCompletion? = nil) {
```

Comment:
I think we can make this private. Also I notice we have unit tests for this method. I think in general, we don't want unit tests for private methods directly - especially if we don't have to. We should instead test the public interface/methods, which then we can assume that our private methods are working correctly.

So in this case, we should test on the `start` method directly.

There's a decent convo about this under `Motivation` in [this article.](https://anthonysciamanna.com/2016/02/14/should-private-methods-be-tested.html)